internal/core/domain/link.go

```diff
@@ -0,0 +1,68 @@
+package domain
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+	core "github.com/iden3/go-iden3-core"
+
+	"github.com/polygonid/sh-id-platform/internal/common"
+)
+
+// CredentialAttributes - credential's attributes
+type CredentialAttributes struct {
+	Name  string `json:"name"`
+	Value string `json:"value"`
+}
+
+// LinkCoreDID - represents a credential offer ID
+type LinkCoreDID core.DID
+
+// Link - represents a credential offer
+type Link struct {
+	ID                       uuid.UUID
+	IssuerDID                LinkCoreDID
+	CreatedAt                time.Time
+	MaxIssuance              *int32
+	ValidUntil               *time.Time
+	SchemaID                 uuid.UUID
+	CredentialExpiration     *time.Time
+	CredentialSignatureProof bool
+	CredentialMTPProof       bool
+	CredentialAttributes     []CredentialAttributes
+	Active                   bool
+}
+
+// NewLink - Constructor
+func NewLink(IssuerDID core.DID, maxIssuance *int32, validUntil *time.Time, schemaID uuid.UUID, CredentialSignatureProof bool, CredentialMTPProof bool, credentialAttributes []CredentialAttributes) *Link {
```

Comment:
```suggestion
func NewLink(IssuerDID core.DID, maxIssuance *int, validUntil *time.Time, schemaID uuid.UUID, CredentialSignatureProof bool, CredentialMTPProof bool, credentialAttributes []CredentialAttributes) *Link {
```
I think it is better to use an int instread of an int32. Less code conversions and probably bigger range.

Reply:
ok